feature/experimental/isis/ate_tests/isis_interface_passive_test/isis_interface_passive_test.go

```diff
@@ -0,0 +1,399 @@
+	isisIntfLevel.GetOrCreateAf(oc.IsisTypes_AFI_TYPE_IPV6, oc.IsisTypes_SAFI_TYPE_UNICAST).Enabled = ygot.Bool(true)
+	isisIntfLevel.GetOrCreateAf(oc.IsisTypes_AFI_TYPE_IPV6, oc.IsisTypes_SAFI_TYPE_UNICAST).Metric = ygot.Uint32(v6_metric)
+
+	// Interface afi-safi configs.
+	intf.GetOrCreateAf(oc.IsisTypes_AFI_TYPE_IPV4, oc.IsisTypes_SAFI_TYPE_UNICAST).Enabled = ygot.Bool(true)
+	intf.GetOrCreateAf(oc.IsisTypes_AFI_TYPE_IPV6, oc.IsisTypes_SAFI_TYPE_UNICAST).Enabled = ygot.Bool(true)
+
+	t.Log("Pushing isis config on DUT")
+	gnmi.Replace(t, dut, configPath.Config(), prot)
+}
+
+// configureATE configures the interfaces and isis protocol on ATE.
+func configureATE(t *testing.T, ate *ondatra.ATEDevice) *ondatra.ATETopology {
+
+	topo := ate.Topology().New()
+	port1 := ate.Port(t, "port1")
+	i1Dut := topo.AddInterface(atePort1attr.Name).WithPort(port1)
+	i1Dut.IPv4().WithAddress(atePort1attr.IPv4CIDR()).WithDefaultGateway(dutPort1Attr.IPv4)
+	i1Dut.IPv6().WithAddress(atePort1attr.IPv6CIDR()).WithDefaultGateway(dutPort1Attr.IPv6)
+
+	isisDut := i1Dut.ISIS()
+	isisDut.
+		WithAreaID(ateAreaAddress).
+		WithTERouterID(atePort1attr.IPv4).
+		WithNetworkTypePointToPoint().
+		WithLevelL2().
+		WithAuthMD5(password).
+		WithAreaAuthMD5(password).
+		WithDomainAuthMD5(password).
+		WithHelloPaddingEnabled(true)
+
+	t.Log("Pushing config to ATE and starting protocols...")
+	topo.Push(t)
+	topo.StartProtocols(t)
+	return topo
+}
+
+// TestIsisInterfacePassive verifies passive isis interface.
+func TestIsisInterfacePassive(t *testing.T) {
+
+	dut := ondatra.DUT(t, "dut")
+	ate := ondatra.ATE(t, "ate")
+	intfName := dut.Port(t, "port1").Name()
+
+	// Configure interface on the DUT.
+	configureDUT(t)
+
+	// Configure network Instance type on DUT.
+	dutConfNIPath := gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut))
+	t.Log("Pushing network Instance type config on DUT")
+	gnmi.Replace(t, dut, dutConfNIPath.Type().Config(), oc.NetworkInstanceTypes_NETWORK_INSTANCE_TYPE_DEFAULT_INSTANCE)
+
+	// Configure isis on DUT.
+	configureIsisDut(t, dut, intfName, dutAreaAddress, dutSysID)
+
+	// Configure interface,isis and traffic on ATE.
+	configureATE(t, ate)
+
+	if deviations.ExplicitInterfaceInDefaultVRF(dut) {
+		intfName = intfName + ".0"
+	}
+	statePath := gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut)).Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_ISIS, ISISInstance).Isis()
+
+	t.Run("Isis telemetry", func(t *testing.T) {
+		t.Run("Verifying adjacency", func(t *testing.T) {
+			adjacencyPath := statePath.Interface(intfName).Level(2).AdjacencyAny().AdjacencyState().State()
+
+			_, ok := gnmi.WatchAll(t, dut, adjacencyPath, time.Minute, func(val *ygnmi.Value[oc.E_Isis_IsisInterfaceAdjState]) bool {
+				state, present := val.Val()
+				return present && state == oc.Isis_IsisInterfaceAdjState_UP
+			}).Await(t)
+			if !ok {
+				t.Logf("Interface %v has no level 2 isis adjacency", intfName)
+				t.Fatal("No isis adjacency reported.")
```

Comment:
no need to both log and fail. If there is important information in the log, put it in the failure message

Reply:
taken care in latest commit